keps/sig-node/2000-graceful-node-shutdown/README.md

```diff
@@ -0,0 +1,843 @@
+implementation on the `shutdown` branch.
+
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+*   Kubelet does not receive shutdown event or is able to create inhibitor lock
+    *   Mitigation: Kubelet does not provide graceful shutdown to pods (same as
+        today’s existing behavior)
+*   Kublet is unable to update `InhibitDelayMaxSec` in logind to match that of
+    `kubeletConfig.ShutdownGracePeriod`
+    *   If there are multiple logind configuration file overrides in
+        `/etc/systemd/logind.conf.d/`, logind will use the config file with the
+        lexicographically latest name. As a result in rare cases, the kubelet’s
+        InhibitDelayMaxSec conf file override may be overwritten by another
+        config file (possibly placed by another service on the machine).
+    *   Mitigation: Kubelet will use current value of `InhibitDelayMaxSec` from
+        logind as the shutdown period which may be less than
+        `kubeletConfig.ShutdownGracePeriod`.
+*   OS / Distro does not use systemd 183
+    *   Mitigation: Kubelet will not provide graceful shutdown to pods (same as
+        today’s existing behavior).
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation, and anything particularly
+challenging to test, should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+*   Unit tests for kubelet of handling shutdown event
+*   New E2E tests to validate node graceful shutdown (note limitation that K8S
```

Comment:
What cloud platforms are you going to test these on?

What systemd versions are you going to test on (likely only the ones provided by base images)?

Reply:
> What cloud platforms are you going to test these on?

Since the e2e tests only currently run on GCE, at least for e2e tests will be limited to GCP. We can run some manual tests on other cloud providers as a sanity check though.

> What systemd versions are you going to test on (likely only the ones provided by base images)?

As you mentioned, it'll be the same versions as provided in the base images the tests is running on.